README.md

```diff
@@ -117,6 +119,10 @@ withdraw_confirm = { gas = 3000000, gas_price = 1000000000 }
 - `home/foreign.poll_interval` - specify how often home node should be polled for changes (in seconds, default: **1**)
 - `home/foreign.request_timeout` - specify request timeout (in seconds, default: **3600**)
 - `home/foreign.password` - path to the file containing a password for the validator's account (to decrypt the key from the keystore)
+- `home/foreign.gas_price_oracle_url` - the URL used to query the current gas price for the home and foreign nodes.
+- `home/foreign.gas_price_timeout` - the number of seconds to wait for an HTTP response from the gas price oracle before using the default gas price.
+- `home/foreign.gas_price_speed_type` - use the gas price from the oracle URL corresponding to this transaction speed.
```

Comment:
Just a suggestion from the plan I had for this feature when I thought I will be working on it. Instead of hard-coding the fact that we'll look up a particular field, I was planning to use [jmespath](https://crates.io/crates/jmespath) crate so that a [JMESPath](http://jmespath.org) expression can be used instead of a single key.

The beauty of this approach was that by default it still is just like having a key with the "speed type" but it allows for more complex expressions should they be needed in the future.

Totally not a requirement for this implementation, but I thought it was a really nice idea. We can add it later as well (however, I'd recommend changing `gas_price_speed_type` to something more generically named in that case, like `gas_price_oracle_expr` or something)

Reply:
Ok, so you are saying that it would be be more flexible to allow the user to specify a JMESPath expression in the config .toml file to allow querying of Gas Price Oracle apis who JSON response schema is different that that of `http://gasprice.poa.network`?

Reply:
Well, that was the idea -- and it will help avoiding the bulk of code for hardcoding the types. `jmespath`'s API would allow to run any JMESPath query over fetched JSON in just a few lines and will make this much more flexible.

Reply:
Let's think how difficult it is to implement support of different schema and how many new dependencies it will introduce? @DrPeterVanNostrand could you clarify?

Reply:
Adding the `jsmespath` crate will add 2 new dependencies `jmespath` and `slug` (`jmespath` also depends on the `serde`, `serde_json`, and `lazy_static` crates, but we already depend on those).

Allowing the user to add a `gas_price_oracle_expr` option to the config TOML file would require changing the `load::Config` and `Config` structs (in bridge::config.rs) to accept the new config option, it would also require parsing the user provided expression String into a `jmespath::Expression` using:

```
let oracle_price_expr = jmespath::compile(USER_PROVIDED_EXPR).map_err(|_| ... )?;
```

If compiling the user provided expression into a `jmespath::Expression` fails, we would either `panic!` or default to some gas price.

Right now, the `GasPriceClient` deserializes HTTP response bodies from the gas-price Oracle into a structure using a constant JSON schema using `serde_json`, we could instead use JMESPath to deserialize the HTTP responses into a generic JSON object (of unknown structure) and use the user provided JMESPath expression to perform a search on it, which might look something like the following:

```
let json_str = /* Use hyper to GET the Oracle's data, then string-ify the response body */
let resp_json = jmespath::Variable::from_json(json_str).map_err(|_| ... )?;
let gas_price = config.<home/foreign>.gas_price_expr
        .search(resp_json).unwrap_or(config.<home/foreign>.default_gas_price);
```

I don't think that using `jmespath` would add a significant ammount of complexity. However, in the case that the user provides a valid JMESPath expression, but that expression does not match any key-value pair in the JSON object returned from the Oracle, we would probably have to default to a gas price to avoid repeatedly querying the Oracle for a response that will never match the JMESPath expression.